hoot-rnd/src/main/cpp/hoot/rnd/language/translators/HootServicesTranslationInfoClient.cpp

```diff
@@ -0,0 +1,169 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. DigitalGlobe
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2018 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+
+#include "HootServicesTranslationInfoClient.h"
+
+// hoot
+#include <hoot/core/util/HootException.h>
+#include <hoot/core/util/Log.h>
+#include <hoot/core/util/ConfigOptions.h>
+#include <hoot/core/util/StringUtils.h>
+#include <hoot/core/util/Factory.h>
+
+// Qt
+#include <QEventLoop>
+#include <QVariant>
+#include <QNetworkRequest>
+#include <QNetworkReply>
+
+// std
+#include <iostream>
+#include <sstream>
+
+namespace hoot
+{
+
+HOOT_FACTORY_REGISTER(TranslationInfoProvider, HootServicesTranslationInfoClient)
+
+HootServicesTranslationInfoClient::HootServicesTranslationInfoClient()
+{
+  _client.reset(new QNetworkAccessManager());
+}
+
+void HootServicesTranslationInfoClient::setConfiguration(const Settings& conf)
+{
+  ConfigOptions opts(conf);
+
+  _translator = opts.getLanguageTranslationHootServicesTranslator();
+  _detectors = opts.getLanguageTranslationHootServicesDetectors();
+  _detectableUrl = opts.getLanguageTranslationHootServicesDetectableLanguagesEndpoint();
+  _translatableUrl = opts.getLanguageTranslationHootServicesTranslatableLanguagesEndpoint();
+  _detectorsUrl = opts.getLanguageTranslationHootServicesDetectorsEndpoint();
+  _translatorsUrl = opts.getLanguageTranslationHootServicesTranslatorsEndpoint();
+}
+
+boost::shared_ptr<boost::property_tree::ptree> HootServicesTranslationInfoClient::getAvailableApps(
+ const QString type)
+{
+ LOG_DEBUG("Checking apps available for: " << type << "...");
+
+ QString urlStr;
+ if (type == "translators")
+ {
+   urlStr = _translatorsUrl;
+ }
+ else
+ {
+   urlStr = _detectorsUrl;
+ }
+ LOG_VARD(urlStr);
+
+ //create the request
+ QUrl url(urlStr);
+ QNetworkRequest request(url);
+
+ //send the request and wait for a response
+ QNetworkReply* reply = _client->get(request);
+ QEventLoop loop;
+ QObject::connect(reply, SIGNAL(finished()), &loop, SLOT(quit()));
+ loop.exec();
+
+ //check for a response error
+ if (reply->error() != QNetworkReply::NoError)
+ {
+   throw HootException(QString("Reply error:\n%1").arg(reply->error()));
+ }
+
```

Comment:
Checkout `core/io/HootNetworkRequest.h` for a class that does all of this already.  Essentially you can replace from line 87-102 with:
```
  //  Do HTTP GET request
  HootNetworkRequest request;
  request.networkRequest(_url);
  return StringUtils::jsonStringToPropTree(QString::fromUtf8(request.getResponseContent().data()));
```

Reply:
cool, I'll check that class out.  thanks.

Reply:
I noticed in some other places that you set some headers, the class could be expanded to add those headers in before making the request.

Reply:
will handle in #2606